### DISCUSSION TITLE
Getting There was a problem with /shipping/profile : Postal code is invalid while updating Listing

### DISCUSSION THREAD

**Discussion by uzairfiaz-gelato** [etsy/open-api#879] [Q&A]
Hi,

When trying to link Shipping Profile with a Listing we are getting 
`{"errors":true,"status":400,"response":{},"body":{"error":"There was a problem with \/shipping\/profile : Postal code is invalid","error_description":""},"bodyArray":{"error":"There was a problem with \/shipping\/profile : Postal code is invalid","error_description":""},"link":null,"exception":{}}`

although the shipping profile created is correct and the postal code is also valid. Can you please look into this? It happens from time to time and randomly. It works correctly sometimes even with the same Shipping postal code and country and sometimes it doesn't.

Some examples:
ShippingId: 188728097317
ListingId: 1406172730

ShippingId: 188728169433
ListingId: 1407599936

ShippingId: 191292541052
ListingId: 1422769431

Thanks in advance.

**Comment by uzairfiaz-gelato** [comment]
Hi @nateswart can you help out here?

Sometimes API does not allow postal code for countries for which it is required in documents e.g CA
Request:
`POST request /v3/application/shops/38936934/shipping-profiles: {"title":"Test","origin_country_iso":"CA","primary_cost":20,"secondary_cost":10,"min_processing_time":2,"max_processing_time":5,"destination_country_iso":"CA","destination_region":null,"origin_postal_code":"B2z 1j6","min_delivery_days":1,"max_delivery_days":6}`

Response:
`{"errors":true,"status":400,"response":{},"body":{"error":"We currently do not support submitting postal codes for this country: CA","error_description":""},"bodyArray":{"error":"We currently do not support submitting postal codes for this country: CA","error_description":""},"link":null,"exception":{}}`

**Reply by Gareth064** [reply]
@uzairfiaz-gelato I think in this instance you will be better to email the Etsy Dev team (developer@etsy.com) to ask for support with this error. I have been told that this is now a "Community Support" repo and so Etsy Dev's are very unlikely to answer many of these issues and discussions anymore. 

If you do email them and get an answer to your problem, then of course it would be great if you shared the answer here so the community would benefit.

**Reply by uzairfiaz-gelato** [reply]
Thank you @Gareth064 will do.

**Reply by emergingdzns** [reply]
Etsy does validate zip codes now for most countries I believe. So if you provide one, it needs to be valid. If the country doesn't require a zip code, try not providing it. I seem to recall that this even happened when the zip code was entered prior to the validation so when you go to update the record, it tries to validate the zip code.

**Reply by uzairfiaz-gelato** [reply]
@emergingdzns I am validating the zip codes and the zip code are valid. I even verify by updating the shipping profile after being created. But for some reason randomly the same zip code is not valid again.
From my above example `B2Z 1J6` is valid zip code for CA and it was created successfully before but now when I try to update the same shipping profile it gives an error.

**Reply by emergingdzns** [reply]
If I remember correctly it will also validate all shipping destinations tied to that profile at the same time. Not sure if that helps, but hopefully it does.

**Reply by uzairfiaz-gelato** [reply]
but according to the documentation, the destination does not take in any postal code parameter. So that should not be the problem or if there is some problem with destinations then the error message returned is incorrect
